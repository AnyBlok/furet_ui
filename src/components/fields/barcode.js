/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import {defineComponent} from '../factory'
import {fields} from './fields';
import VueBarcode from 'vue-barcode';

defineComponent('furet-ui-list-field-barcode', {
  template:`
    <span v-if="isHidden" />
    <barcode v-else v-bind:value="value" v-bind="config.options">
      Can't generate the barcode
    </barcode>
  `,
  extend: ['furet-ui-list-field-common'],
  prototype: {
    components: {
      barcode: VueBarcode
    },
  },
})
fields.list.barcode = 'furet-ui-list-field-barcode'


defineComponent('furet-ui-form-field-barcode', {
  template: `
    <furet-ui-form-field-common-tooltip-field
      v-bind:resource="resource"
      v-bind:data="data"
      v-bind:config="config"
    >
      <barcode v-if="isReadonly" v-bind:value="value" v-bind="config.options">
        Can't generate the barcode
      </barcode>
      <b-input 
        v-else
        v-bind:value="value" 
        v-on:input="updateValue"
        v-bind:maxlength="config.maxlength"
        v-bind:placeholder="config.placeholder"
        icon-pack="fa"
        v-bind:icon="config.icon"
        v-bind:key="config.key"
      />
    </furet-ui-form-field-common-tooltip-field>
  `,
  extend: ['furet-ui-form-field-common'],
  prototype: {
    components: {
      barcode: VueBarcode
    },
  },
})
fields.form.barcode = 'furet-ui-form-field-barcode'
