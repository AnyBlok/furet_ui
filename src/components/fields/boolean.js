// /**
// This file is a part of the FuretUI project
// 
//    Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>
// 
// This Source Code Form is subject to the terms of the Mozilla Public License,
// v. 2.0. If a copy of the MPL was not distributed with this file,You can
// obtain one at http://mozilla.org/MPL/2.0/.
// **/
// import _ from 'underscore';
import {defineComponent} from '../factory'
import {fields} from './fields';
import { listTemplate } from './common';


defineComponent('furet-ui-list-field-boolean', {
  template: `
    <div>
      <span v-if="isHidden" />
      <b-checkbox 
        v-else
        v-model="checked"
        disabled
      />
    </div>`,
  extend: ['furet-ui-list-field-common'],
  prototype: {
    computed: {
      checked () {
        return eval(this.value) ? true : false;
      },
    },
  },
})
fields.list.boolean = 'furet-ui-list-field-boolean'


defineComponent('furet-ui-list-field-yesno', {
  template: listTemplate,
  extend: ['furet-ui-list-field-common'],
  prototype: {
    computed: {
      value () {
        const base = 'components.fields.yesno'
        return this.$t(eval(this.data[this.config.name] || '') ? `${base}.yes` : `${base}.no`);
      },
    },
  },
})
fields.list.yesno = 'furet-ui-list-field-yesno'

//export const FieldThumbnailBoolean = Vue.component('furet-ui-thumbnail-field-boolean', {
//    mixins: [ThumbnailMixin],
//    template: `
//        <div v-if="this.isHidden" />
//        <b-tooltip 
//            v-bind:label="getTooltip" 
//            v-bind:position="tooltipPosition"
//            v-else
//        >
//            <b-checkbox 
//                v-model="value" 
//                disabled
//            >
//                {{this.label}}
//            </b-checkbox>
//        </b-tooltip>`,
//    computed: {
//        value () {
//            return eval(this.data[this.name]) ? true : false;
//        },
//    }
//})


defineComponent('furet-ui-form-field-boolean', {
  template: `
    <furet-ui-form-field-common-tooltip
      v-bind:data="data"
      v-bind:config="config"
    >
      <b-checkbox 
        v-model="checked" 
        v-bind:disabled="isReadonly"
      >
        {{ $t(config.label) }}
      </b-checkbox>
    </furet-ui-form-field-common-tooltip>`,
  extend: ['furet-ui-form-field-common'],
  prototype: {
    computed: {
      checked: {
        get () {
          return eval(this.value) ? true : false;
        },
        set (value) {
          this.updateValue (value);
        },
      },
    },
  },
})
fields.form.boolean = 'furet-ui-form-field-boolean'
