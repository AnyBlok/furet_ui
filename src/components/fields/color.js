/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import {defineComponent} from '../factory'
import {fields} from './fields';

defineComponent('furet-ui-list-field-color', {
  template:`
    <span v-if="isHidden" />
    <component 
      v-else-if="config.slot" 
      v-bind:is="component_template"
      v-bind:config="config"
      v-bind:resource="resource"
      v-bind:data="data"
      v-bind:value="value"
    />
    <b-input 
      v-else-if="value"
      type="color"
      v-bind:value="value" 
      disabled
    />
    <span v-else />
  `,
  extend: ['furet-ui-list-field-common'],
})
fields.list.color = 'furet-ui-list-field-color'

defineComponent('furet-ui-thumbnail-field-color', {
  template: `
    <furet-ui-thumbnail-field-common-tooltip-field
      v-bind:resource="resource"
      v-bind:data="data"
      v-bind:config="config"
    >
      <component 
        v-if="config.slot" 
        v-bind:is="component_template"
        v-bind:config="config"
        v-bind:resource="resource"
        v-bind:data="data"
        v-bind:value="value"
      />
      <b-input 
        v-else-if="value"
        type="color"
        v-bind:value="value" 
        disabled
      />
    </furet-ui-thumbnail-field-common-tooltip-field>
  `,
  extend: ['furet-ui-thumbnail-field-common'],
})
fields.thumbnail.color = 'furet-ui-thumbnail-field-color'
fields.kanban.color = 'furet-ui-thumbnail-field-color'

defineComponent('furet-ui-form-field-color', {
  template: `
    <furet-ui-form-field-common-tooltip-field
      v-bind:resource="resource"
      v-bind:data="data"
      v-bind:config="config"
    >
      <div class="field has-addons">
        <p v-if="isReadonly" class="control is-expanded">
          <component 
            v-if="config.slot" 
            v-bind:is="component_template"
            v-bind:config="config"
            v-bind:resource="resource"
            v-bind:data="data"
            v-bind:value="value"
          />
          <b-input 
            v-else-if="data"
            type="color"
            v-bind:value="value" 
            disabled
          >
          </b-input>
        </p>
        <p v-else class="control">
          <b-input 
            style="width: 100px"
            v-if="data || !isReadonly"
            type="color"
            v-bind:value="value" 
            v-on:input="updateValue"
            v-bind:disabled="isReadonly"
          >
          </b-input>
        </p>
        <p class="control is-expanded">
          <b-input 
            v-if="!isReadonly"
            v-bind:value="value" 
            v-on:input="updateValue"
            v-bind:placeholder="config.placeholder"
            icon-pack="fa"
            v-bind:icon="config.icon"
            pattern="#[A-Fa-f1-9]{6}"
          />
        </p>
      </div>
    </furet-ui-form-field-common-tooltip-field>
  `,
  extend: ['furet-ui-form-field-common'],
})
fields.form.color = 'furet-ui-form-field-color'
