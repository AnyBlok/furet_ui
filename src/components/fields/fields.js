export const fields = {
  list: {},
  form: {},
  thumbnail: {},
  kanban: {},
};
export default {
    fields,
}
