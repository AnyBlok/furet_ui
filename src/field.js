/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import React from 'react';
import plugin from './plugin';

/**
 * return the component for viewType and fieldType
**/
export const getField = (viewType, fieldType, attribs, value) => {
    let field = plugin.get(['field', viewType, fieldType]);
    if (!field) {
        field = plugin.get(['field', 'Unknown']);
        console.log('getField', viewType, fieldType, attribs, value)
    }
    const id = 'view-type-' + viewType + '-field_type-' + fieldType + '-field-name-' + (attribs.name || '');
    return React.createElement(field, Object.assign({id}, attribs, {value}));
}

export default {
    getField,
}
