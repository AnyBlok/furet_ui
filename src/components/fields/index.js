import { fields } from './fields';
import './common';
import './boolean';
import './datetime';
import './integer';
import './password';
import './timestamp';
import './selection';
import './string';

import './relationship';

import { defineComponent } from '../factory';

defineComponent('furet-ui-unknown-field', {
  template : `<div>Unknow Field <strong>{{config.type}}</strong></div>`,
  prototype: {
    props: ['config', 'data'],
  }
})

defineComponent('furet-ui-field', {
  template : `
    <component 
      v-bind:is="component" 
      v-bind:resource="resource" 
      v-bind:config="config" 
      v-bind:data="data"
    />
  `,
  prototype: {
    props: ['resource', 'config', 'data'],
    computed: {
      component () {
        return (fields[this.resource.type] || {})[this.config.type] || 'furet-ui-unknown-field'
      },
    },
  }
})
