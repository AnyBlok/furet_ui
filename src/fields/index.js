/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import React from 'react';
import plugin from '../plugin';
import './string';
import './selection';
import './date';
import './integer';
import './float';
import './url';
import './uuid';
import './sequence';
import './password';
import './color';
import './text';

/**
 * Unknown field, used only if the wanted field is not available
**/
plugin.set(['field'], {Unknown: (props) => {
    return <div>{props.value}</div>
}});

/**
 * return the component for viewType and fieldType
**/
export const getField = (viewType, fieldType, attribs, value) => {
    let field = plugin.get(['field', viewType, fieldType]);
    if (!field) {
        field = plugin.get(['field', 'Unknown']);
        console.log('getField', viewType, fieldType, attribs, value)
    }
    const id = 'view-type-' + viewType + '-field_type-' + fieldType + '-field-name-' + (attribs.name || '');
    return React.createElement(field, Object.assign({id}, attribs, {value}));
}

export default {
    getField,
}
