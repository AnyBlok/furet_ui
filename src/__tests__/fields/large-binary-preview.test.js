/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import React from 'react';
import renderer from 'react-test-renderer';
import '../../fields'
import {getField} from '../../field';

test('getField for List', () => {
    const component = renderer.create(getField('List', 'LargeBinaryPreview', {data: {}}, 'Test'));
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
test('getField for List 2', () => {
    const component = renderer.create(getField('List', 'LargeBinaryPreview', {data: {}}, null));
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
test('getField for List with filename', () => {
    const component = renderer.create(getField('List', 'LargeBinaryPreview', {data: {}, filename: 'file_name'}, 'Test'));
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
test('getField for List with filename 2', () => {
    const component = renderer.create(getField('List', 'LargeBinaryPreview', {data: {file_name: 'File name'}, filename: 'file_name'}, 'Test'));
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
test('getField for Thumbnail', () => {
    const component = renderer.create(getField('Thumbnail', 'LargeBinaryPreview', {data: {}}, 'Test'));
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
test('getField for Thumbnail 2', () => {
    const component = renderer.create(getField('Thumbnail', 'LargeBinaryPreview', {data: {}}, null));
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
test('getField for Thumbnail with filename', () => {
    const component = renderer.create(getField('Thumbnail', 'LargeBinaryPreview', {data: {}, filename: 'file_name'}, 'Test'));
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
test('getField for Thumbnail with filename 2', () => {
    const component = renderer.create(getField('Thumbnail', 'LargeBinaryPreview', {data: {file_name: 'File name'}, filename: 'file_name'}, 'Test'));
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
test('getField for Form', () => {
    const component = renderer.create(getField('Form', 'LargeBinaryPreview', {data: {}}, 'Test'));
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
test('getField for Form 2', () => {
    const component = renderer.create(getField('Form', 'LargeBinaryPreview', {data: {}}, null));
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
test('getField for Form readonly', () => {
    const component = renderer.create(getField('Form', 'LargeBinaryPreview', {data: {}, readonly: true, label: 'Test'}, 'Test'));
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
test('getField for Form readonly 2', () => {
    const component = renderer.create(getField('Form', 'LargeBinaryPreview', {data: {}, readonly: true, label: 'Test'}, null));
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('getField for Form required', () => {
    const component = renderer.create(getField('Form', 'LargeBinaryPreview', {data: {}, required: '1', label: 'Test'}, 'Test'));
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('getField for Form required with empty value', () => {
    const component = renderer.create(getField('Form', 'LargeBinaryPreview', {data: {}, required: '1', label: 'Test'}, ''));
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
test('getField for Form with filename', () => {
    const component = renderer.create(getField('Form', 'LargeBinaryPreview', {data: {}, filename: 'file_name'}, 'Test'));
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
test('getField for Form with filename 2', () => {
    const component = renderer.create(getField('Form', 'LargeBinaryPreview', {data: {}, filename: 'file_name'}, null));
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
test('getField for Form with filename 3', () => {
    const component = renderer.create(getField('Form', 'LargeBinaryPreview', {data: {file_name: 'File name'}, filename: 'file_name'}, 'Test'));
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
test('getField for Form with filename 4', () => {
    const component = renderer.create(getField('Form', 'LargeBinaryPreview', {data: {file_name: 'File name'}, filename: 'file_name'}, null));
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
test('getField for Form with filename and readonly', () => {
    const component = renderer.create(getField('Form', 'LargeBinaryPreview', {readonly: true, data: {file_name: 'File name'}, filename: 'file_name'}, 'Test'));
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
test('getField for Form with filename and readonly 2', () => {
    const component = renderer.create(getField('Form', 'LargeBinaryPreview', {readonly: true, data: {file_name: 'File name'}, filename: 'file_name'}, null));
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
