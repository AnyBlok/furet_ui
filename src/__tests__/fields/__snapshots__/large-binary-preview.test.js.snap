// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getField for Form 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="view-type-Form-field_type-LargeBinaryPreview-field-name-"
    style={Object {}}
  >
     
  </label>
  <div
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <img
      alt=""
      className="img-responsive img-thumbnail"
      src="Test"
      style={
        Object {
          "opacity": 1,
        }
      }
    />
    <div
      style={
        Object {
          "left": "50%",
          "opacity": 0,
          "position": "absolute",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "transition": ".5s ease",
        }
      }
    >
      <ul
        className="list-inline"
      >
        <li>
          <div>
            <label
              htmlFor="view-type-Form-field_type-LargeBinaryPreview-field-name-"
            >
              <a>
                <i
                  className="fa fa-upload fa-lg"
                />
              </a>
            </label>
            <input
              className="form-control"
              disabled={undefined}
              id="view-type-Form-field_type-LargeBinaryPreview-field-name-"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
              type="file"
            />
          </div>
        </li>
        <li>
          <a
            onClick={[Function]}
          >
            <i
              className="fa fa-trash fa-lg"
            />
          </a>
        </li>
        <li>
          <a
            download=""
            href="Test"
            target="_blank"
          >
            <i
              className="fa fa-download fa-lg"
            />
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`getField for Form 2 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="view-type-Form-field_type-LargeBinaryPreview-field-name-"
    style={Object {}}
  >
     
  </label>
  <div>
    <label
      htmlFor="view-type-Form-field_type-LargeBinaryPreview-field-name-"
    >
      <a>
        <i
          className="fa fa-upload fa-lg"
        />
      </a>
    </label>
    <input
      className="form-control"
      disabled={undefined}
      id="view-type-Form-field_type-LargeBinaryPreview-field-name-"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
  </div>
</div>
`;

exports[`getField for Form readonly 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="view-type-Form-field_type-LargeBinaryPreview-field-name-"
    style={Object {}}
  >
    Test
  </label>
  <div
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <img
      alt=""
      className="img-responsive img-thumbnail"
      src="Test"
      style={
        Object {
          "opacity": 1,
        }
      }
    />
    <div
      style={
        Object {
          "left": "50%",
          "opacity": 0,
          "position": "absolute",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "transition": ".5s ease",
        }
      }
    >
      <ul
        className="list-inline"
      >
        <li>
          <a
            download=""
            href="Test"
            target="_blank"
          >
            <i
              className="fa fa-download fa-lg"
            />
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`getField for Form readonly 2 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="view-type-Form-field_type-LargeBinaryPreview-field-name-"
    style={Object {}}
  >
    Test
  </label>
</div>
`;

exports[`getField for Form required 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="view-type-Form-field_type-LargeBinaryPreview-field-name-"
    style={
      Object {
        "color": "#3f51b5",
      }
    }
  >
    Test
  </label>
  <div
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <img
      alt=""
      className="img-responsive img-thumbnail"
      src="Test"
      style={
        Object {
          "opacity": 1,
        }
      }
    />
    <div
      style={
        Object {
          "left": "50%",
          "opacity": 0,
          "position": "absolute",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "transition": ".5s ease",
        }
      }
    >
      <ul
        className="list-inline"
      >
        <li>
          <div>
            <label
              htmlFor="view-type-Form-field_type-LargeBinaryPreview-field-name-"
            >
              <a>
                <i
                  className="fa fa-upload fa-lg"
                />
              </a>
            </label>
            <input
              className="form-control"
              disabled={undefined}
              id="view-type-Form-field_type-LargeBinaryPreview-field-name-"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
              type="file"
            />
          </div>
        </li>
        <li>
          <a
            onClick={[Function]}
          >
            <i
              className="fa fa-trash fa-lg"
            />
          </a>
        </li>
        <li>
          <a
            download=""
            href="Test"
            target="_blank"
          >
            <i
              className="fa fa-download fa-lg"
            />
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`getField for Form required with empty value 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="view-type-Form-field_type-LargeBinaryPreview-field-name-"
    style={
      Object {
        "color": "#3f51b5",
      }
    }
  >
    Test
  </label>
  <div>
    <label
      htmlFor="view-type-Form-field_type-LargeBinaryPreview-field-name-"
    >
      <a>
        <i
          className="fa fa-upload fa-lg"
        />
      </a>
    </label>
    <input
      className="form-control"
      disabled={undefined}
      id="view-type-Form-field_type-LargeBinaryPreview-field-name-"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
  </div>
  <div>
    <span
      style={
        Object {
          "color": "#f44336",
        }
      }
    >
      This field is required
    </span>
  </div>
</div>
`;

exports[`getField for Form with filename 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="view-type-Form-field_type-LargeBinaryPreview-field-name-"
    style={Object {}}
  >
     
  </label>
  <div
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <img
      alt=""
      className="img-responsive img-thumbnail"
      src="Test"
      style={
        Object {
          "opacity": 1,
        }
      }
    />
    <div
      style={
        Object {
          "left": "50%",
          "opacity": 0,
          "position": "absolute",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "transition": ".5s ease",
        }
      }
    >
      <ul
        className="list-inline"
      >
        <li>
          <div>
            <label
              htmlFor="view-type-Form-field_type-LargeBinaryPreview-field-name-"
            >
              <a>
                <i
                  className="fa fa-upload fa-lg"
                />
              </a>
            </label>
            <input
              className="form-control"
              disabled={undefined}
              id="view-type-Form-field_type-LargeBinaryPreview-field-name-"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
              type="file"
            />
          </div>
        </li>
        <li>
          <a
            onClick={[Function]}
          >
            <i
              className="fa fa-trash fa-lg"
            />
          </a>
        </li>
        <li>
          <a
            download=""
            href="Test"
            target="_blank"
          >
            <i
              className="fa fa-download fa-lg"
            />
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`getField for Form with filename 2 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="view-type-Form-field_type-LargeBinaryPreview-field-name-"
    style={Object {}}
  >
     
  </label>
  <div>
    <label
      htmlFor="view-type-Form-field_type-LargeBinaryPreview-field-name-"
    >
      <a>
        <i
          className="fa fa-upload fa-lg"
        />
      </a>
    </label>
    <input
      className="form-control"
      disabled={undefined}
      id="view-type-Form-field_type-LargeBinaryPreview-field-name-"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
  </div>
</div>
`;

exports[`getField for Form with filename 3 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="view-type-Form-field_type-LargeBinaryPreview-field-name-"
    style={Object {}}
  >
     
  </label>
  <div
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <img
      alt="File name"
      className="img-responsive img-thumbnail"
      src="Test"
      style={
        Object {
          "opacity": 1,
        }
      }
    />
    <div
      style={
        Object {
          "left": "50%",
          "opacity": 0,
          "position": "absolute",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "transition": ".5s ease",
        }
      }
    >
      <ul
        className="list-inline"
      >
        <li>
          <div>
            <label
              htmlFor="view-type-Form-field_type-LargeBinaryPreview-field-name-"
            >
              <a>
                <i
                  className="fa fa-upload fa-lg"
                />
              </a>
            </label>
            <input
              className="form-control"
              disabled={undefined}
              id="view-type-Form-field_type-LargeBinaryPreview-field-name-"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
              type="file"
            />
          </div>
        </li>
        <li>
          <a
            onClick={[Function]}
          >
            <i
              className="fa fa-trash fa-lg"
            />
          </a>
        </li>
        <li>
          <a
            download="File name"
            href="Test"
            target="_blank"
          >
            <i
              className="fa fa-download fa-lg"
            />
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`getField for Form with filename 4 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="view-type-Form-field_type-LargeBinaryPreview-field-name-"
    style={Object {}}
  >
     
  </label>
  <div>
    <label
      htmlFor="view-type-Form-field_type-LargeBinaryPreview-field-name-"
    >
      <a>
        <i
          className="fa fa-upload fa-lg"
        />
      </a>
    </label>
    <input
      className="form-control"
      disabled={undefined}
      id="view-type-Form-field_type-LargeBinaryPreview-field-name-"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
  </div>
</div>
`;

exports[`getField for Form with filename and readonly 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="view-type-Form-field_type-LargeBinaryPreview-field-name-"
    style={Object {}}
  >
     
  </label>
  <div
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <img
      alt="File name"
      className="img-responsive img-thumbnail"
      src="Test"
      style={
        Object {
          "opacity": 1,
        }
      }
    />
    <div
      style={
        Object {
          "left": "50%",
          "opacity": 0,
          "position": "absolute",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "transition": ".5s ease",
        }
      }
    >
      <ul
        className="list-inline"
      >
        <li>
          <a
            download="File name"
            href="Test"
            target="_blank"
          >
            <i
              className="fa fa-download fa-lg"
            />
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`getField for Form with filename and readonly 2 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="view-type-Form-field_type-LargeBinaryPreview-field-name-"
    style={Object {}}
  >
     
  </label>
</div>
`;

exports[`getField for List 1`] = `
<div>
  <a
    download=""
    href="Test"
    onClick={[Function]}
    target="_blank"
    title=""
  >
    <img
      alt=""
      className="img-responsive img-thumbnail"
      src="Test"
    />
  </a>
</div>
`;

exports[`getField for List 2 1`] = `
<div>
  <a
    download=""
    href=""
    onClick={[Function]}
    target="_blank"
    title=""
  >
    <img
      alt=""
      className="img-responsive img-thumbnail"
      src=""
    />
  </a>
</div>
`;

exports[`getField for List with filename 1`] = `
<div>
  <a
    download=""
    href="Test"
    onClick={[Function]}
    target="_blank"
    title=""
  >
    <img
      alt=""
      className="img-responsive img-thumbnail"
      src="Test"
    />
  </a>
</div>
`;

exports[`getField for List with filename 2 1`] = `
<div>
  <a
    download="File name"
    href="Test"
    onClick={[Function]}
    target="_blank"
    title="File name"
  >
    <img
      alt="File name"
      className="img-responsive img-thumbnail"
      src="Test"
    />
  </a>
</div>
`;

exports[`getField for Thumbnail 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="view-type-Thumbnail-field_type-LargeBinaryPreview-field-name-"
  >
     
  </label>
  <div>
    <a
      download=""
      href="Test"
      onClick={[Function]}
      target="_blank"
      title=""
    >
      <img
        alt=""
        className="img-responsive img-thumbnail"
        src="Test"
      />
    </a>
  </div>
</div>
`;

exports[`getField for Thumbnail 2 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="view-type-Thumbnail-field_type-LargeBinaryPreview-field-name-"
  >
     
  </label>
  <div>
    <a
      download=""
      href=""
      onClick={[Function]}
      target="_blank"
      title=""
    >
      <img
        alt=""
        className="img-responsive img-thumbnail"
        src=""
      />
    </a>
  </div>
</div>
`;

exports[`getField for Thumbnail with filename 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="view-type-Thumbnail-field_type-LargeBinaryPreview-field-name-"
  >
     
  </label>
  <div>
    <a
      download=""
      href="Test"
      onClick={[Function]}
      target="_blank"
      title=""
    >
      <img
        alt=""
        className="img-responsive img-thumbnail"
        src="Test"
      />
    </a>
  </div>
</div>
`;

exports[`getField for Thumbnail with filename 2 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="view-type-Thumbnail-field_type-LargeBinaryPreview-field-name-"
  >
     
  </label>
  <div>
    <a
      download="File name"
      href="Test"
      onClick={[Function]}
      target="_blank"
      title="File name"
    >
      <img
        alt="File name"
        className="img-responsive img-thumbnail"
        src="Test"
      />
    </a>
  </div>
</div>
`;
